packages/@aws-cdk/aws-codedeploy/lib/server/application.ts

```diff
@@ -58,53 +34,74 @@ export interface ServerApplicationProps {
-   * @returns a Construct representing a reference to an existing Application
-   */
-  public static import(scope: cdk.Construct, id: string, props: ServerApplicationImportProps): IServerApplication {
+  public static import(scope: cdk.Construct, id: string, props: ServerApplicationImportProps) {
     return new ImportedServerApplication(scope, id, props);
   }
 
   public readonly applicationArn: string;
   public readonly applicationName: string;
+  public readonly computePlatform: ComputePlatform.Server;
 
-  constructor(scope: cdk.Construct, id: string, props?: ServerApplicationProps) {
+  constructor(scope: cdk.Construct, id: string, props: ServerApplicationProps = {}) {
     super(scope, id);
 
     const resource = new CfnApplication(this, 'Resource', {
-      applicationName: props && props.applicationName,
-      computePlatform: 'Server',
+      applicationName: props.applicationName,
+      computePlatform: ComputePlatform.Server,
     });
 
+    this.computePlatform = ComputePlatform.Server;
     this.applicationName = resource.ref;
     this.applicationArn = applicationNameToArn(this.applicationName, this);
   }
 
   public export(): ServerApplicationImportProps {
     return {
-      applicationName: new cdk.Output(this, 'ApplicationName', { value: this.applicationName }).makeImportValue().toString()
+      applicationName: new cdk.Output(this, 'ApplicationName', { value: this.applicationName }).makeImportValue().toString(),
+      computePlatform: this.computePlatform
     };
   }
 }
 
-function applicationNameToArn(applicationName: string, scope: cdk.IConstruct): string {
-  return cdk.Stack.find(scope).formatArn({
-    service: 'codedeploy',
-    resource: 'application',
-    resourceName: applicationName,
-    sep: ':',
-  });
+/**
+ * Properties of a reference to a CodeDeploy Application.
+ *
+ * @see ServerApplication#import
+ * @see ServerApplication#export
+ */
+export interface ServerApplicationImportProps {
+  /**
+   * The physical, human-readable name of the CodeDeploy EC2/on-premise Application we're referencing.
+   * The Application must be in the same account and region as the root Stack.
+   */
+  applicationName: string;
+
+  /**
+   * The compute platform of the Server application, is always Server.
+   */
+  computePlatform?: ComputePlatform.Server;
+}
+
+export class ImportedServerApplication extends cdk.Construct implements IServerApplication {
+  public readonly applicationArn: string;
+  public readonly applicationName: string;
+  public readonly computePlatform: ComputePlatform.Server;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: ServerApplicationImportProps) {
+    super(scope, id);
+    this.applicationName = props.applicationName;
+    this.applicationArn = applicationNameToArn(this.applicationName, this);
+    this.computePlatform = ComputePlatform.Server;
+  }
+
+  public export(): ServerApplicationImportProps {
+    return this.props;
+  }
```

Comment:
I would do no changes to this entire file other than moving it to the `server`directory.

Reply:
Done. I have moved some common functions to `utils.ts`, but no changes apart from that.